libs/ui/map/src/lib/components/map-legend/map-legend.component.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing'
+import { MapLegendComponent } from './map-legend.component'
+import { MapContext } from '@geospatial-sdk/core'
+import { createLegendFromLayer } from '@geospatial-sdk/legend'
+
+jest.mock('@geospatial-sdk/legend', () => ({
+  createLegendFromLayer: jest.fn(),
+}))
+
+describe('MapLegendComponent', () => {
+  let component: MapLegendComponent
+  let fixture: ComponentFixture<MapLegendComponent>
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [MapLegendComponent],
+    }).compileComponents()
+  })
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MapLegendComponent)
+    component = fixture.componentInstance
+    fixture.detectChanges()
+  })
+
+  it('should create', () => {
+    expect(component).toBeTruthy()
+  })
+
+  describe('Change of map-context', () => {
+    it('should not create legend on first change', async () => {
```

Comment:
Why? I feel like the legend should always be shown, regardless of if it's a first change or not